runtime/permissions/prompter.rs

```diff
@@ -66,29 +68,32 @@ pub fn permission_prompt(
   api_name: Option<&str>,
   is_unary: bool,
 ) -> PromptResponse {
-  if let Some(before_callback) = MAYBE_BEFORE_PROMPT_CALLBACK.lock().as_mut() {
-    before_callback();
-  }
-  let stack = MAYBE_CURRENT_STACKTRACE.lock().take();
+  MAYBE_BEFORE_PROMPT_CALLBACK.with(|cb| {
+    if let Some(before_callback) = cb.lock().as_mut() {
+      before_callback();
+    }
+  });
+  let stack = MAYBE_CURRENT_STACKTRACE.with(|t| t.lock().take());
   let r = PERMISSION_PROMPTER
-    .lock()
-    .prompt(message, flag, api_name, is_unary, stack);
-  if let Some(after_callback) = MAYBE_AFTER_PROMPT_CALLBACK.lock().as_mut() {
-    after_callback();
-  }
+    .with(|p| p.lock().prompt(message, flag, api_name, is_unary, stack));
+  MAYBE_AFTER_PROMPT_CALLBACK.with(|cb| {
+    if let Some(after_callback) = cb.lock().as_mut() {
+      after_callback();
+    }
+  });
   r
 }
 
 pub fn set_prompt_callbacks(
   before_callback: PromptCallback,
   after_callback: PromptCallback,
 ) {
-  *MAYBE_BEFORE_PROMPT_CALLBACK.lock() = Some(before_callback);
-  *MAYBE_AFTER_PROMPT_CALLBACK.lock() = Some(after_callback);
+  MAYBE_BEFORE_PROMPT_CALLBACK.with(|cb| *cb.lock() = Some(before_callback));
+  MAYBE_AFTER_PROMPT_CALLBACK.with(|cb| *cb.lock() = Some(after_callback));
```

Comment:
This is now incorrect - if these are thread locals then `set_prompt_callback` needs to be called in each worker thread spawned - you'd need to update `op_create_worker` in `runtime/ops/worker_host.rs`

Reply:
I see! That makes sense. Thanks @bartlomieju!

I'll work on updating `worker_host.rs`. I have some ideas and will test them to see what works best.